### DISCUSSION TITLE
Why not use dot notation for subcomponents?

### DISCUSSION THREAD

**Discussion by hkfi** [shadcn-ui/ui#2256] [Q&A]
I think it would be much better for subcomponents to be used like this:

```
<Card>
  <Card.Header>
    <Card.Header.Title>Card Title</Card.Header.Title>
    <Card.Header.Description>Card Description</Card.Header.Description>
  </Card.Header>
  <Card.Content>
    <p>Card Content</p>
  </Card.Content>
  <Card.Footer>
    <p>Card Footer</p>
  </Card.Footer>
</Card>
```

instead of what we have now:

```
<Card>
  <CardHeader>
    <CardTitle>Card Title</CardTitle>
    <CardDescription>Card Description</CardDescription>
  </CardHeader>
  <CardContent>
    <p>Card Content</p>
  </CardContent>
  <CardFooter>
    <p>Card Footer</p>
  </CardFooter>
</Card>
```

Pros of using dot notation:
- Communicates hierarchy of where a subcomponent can and is intended to be used.
- Better autocomplete as a result of the previous point.

Cons:
- None that I know of.

Are there any reasons why the components are not like this?

**Comment by burakkirkil** [comment]
I also prefer dot notation. However, one of the disadvantages of this method is that it does not support tree shaking. This may be the reason.

**Comment by burakkirkil** [comment]
https://github.com/shadcn-ui/ui/issues/93
